### PR TITLE
Update deprecated ccall syntax in arpack

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

File: base/linalg/arpack.jl [hunk 1]
```diff
@@ -257,30 +258,31 @@ for (T, TR, naupd_name, neupd_name) in
...
+                       rwork::Vector{$TR}, info)
             ccall(($(string(naupd_name)), :libarpack), Void,
-                  (Ptr{BlasInt}, Ptr{UInt8}, Ptr{BlasInt}, Ptr{UInt8}, Ptr{BlasInt},
-                   Ptr{$TR}, Ptr{$T}, Ptr{BlasInt}, Ptr{$T}, Ptr{BlasInt},
-                   Ptr{BlasInt}, Ptr{BlasInt}, Ptr{$T}, Ptr{$T}, Ptr{BlasInt},
-                   Ptr{$TR}, Ptr{BlasInt}),
-                  ido, bmat, &n, evtype, &nev, TOL, resid, &ncv, v, &ldv,
-                  iparam, ipntr, workd, workl, &lworkl, rwork, info)
+                  (Ref{BlasInt}, Ptr{UInt8}, Ref{BlasInt}, Ptr{UInt8}, Ref{BlasInt},
+                   Ptr{$TR}, Ptr{$T}, Ref{BlasInt}, Ptr{$T}, Ref{BlasInt},
```
Comment (musm) [inline base/linalg/arpack.jl]: It seems to me that all of these should be `Ref` not `Ptr`?  From https://docs.julialang.org/en/latest/manual/calling-c-and-fortran-code/#Some-Examples-of-C-Wrappers-1

> Note that for this code to work correctly, result_array must be declared to be of type Ref{Cdouble} and not Ptr{Cdouble}. The memory is managed by Julia and the Ref signature alerts Julia's garbage collector to keep managing the memory for result_array while the ccall executes. If Ptr{Cdouble} were used instead, the ccall may still work, but Julia's garbage collector would not be aware that the memory declared for result_array is being used by the external C function. As a result, the code may produce a memory leak if result_array never gets freed by the garbage collector, or if the garbage collector prematurely frees result_array, the C function may end up throwing an invalid memory access exception.

Comment (yuyichao) [inline base/linalg/arpack.jl]: Doesn't really matter when used correctly.

**Comment by musm** [comment]
part of https://github.com/JuliaLang/julia/issues/6080

**Comment by musm** [comment]
G2g? Maybe benchmarks?

**Comment by fredrikekre** [comment]
Didn't find any ARPACK benchmarks in the suite, but better safe than sorry
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by musm** [comment]
Thanks

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/23307#issuecomment-323599884) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/09fe0dd98bf8e73baccd01437b8af5e8faaaeb2a/7dc80c4_vs_1323561/report.md). cc @ararslan

**Comment by musm** [comment]
g2g?

**Comment by musm** [comment]
part of #6080 (ref)

**Pull request merged**